src/main/java/hudson/plugins/active_directory/ActiveDirectoryUnixAuthenticationProvider.java

```diff
@@ -482,44 +482,49 @@ private String getPrincipalName(String username, String domainName) {
         if (userDN.contains("/")) {
             userDN = userDN.replace("/","\\/");
         }
-        LOGGER.finer("Looking up group of "+userDN);
+        Set<GrantedAuthority> groups = new HashSet<GrantedAuthority>();
+
+        LOGGER.log(Level.FINER, "Looking up group of {0}", userDN);
         Attributes id = context.getAttributes(userDN,new String[]{"tokenGroups","memberOf","CN"});
         Attribute tga = id.get("tokenGroups");
-        if (tga==null) {// see JENKINS-11644. still trying to figure out when this happens
-            LOGGER.warning("Failed to retrieve tokenGroups for "+userDN);
-            HashSet<GrantedAuthority> r = new HashSet<GrantedAuthority>();
-            r.add(new GrantedAuthorityImpl("unable-to-retrieve-tokenGroups"));
-            return r;
-        }
-
-        // build up the query to retrieve all the groups
-        StringBuilder query = new StringBuilder("(|");
-        List<byte[]> sids = new ArrayList<byte[]>();
 
-        NamingEnumeration<?> tokenGroups = tga.getAll();
-        while (tokenGroups.hasMore()) {
-            byte[] gsid = (byte[])tokenGroups.next();
-            query.append("(objectSid={"+sids.size()+"})");
-            sids.add(gsid);
+        if (tga==null) {
+            // tga will be null if you are not using a global catalogue
+            // or if the user is not actually a member of any security groups.
+            LOGGER.log(Level.FINE, "Failed to retrieve tokenGroups for {0}", userDN);
+            // keep on trucking as we can still use memberOf for Distribution Groups.
         }
-        tokenGroups.close();
-
-        query.append(")");
-
-        Set<GrantedAuthority> groups = new HashSet<GrantedAuthority>();
-
-        NamingEnumeration<SearchResult> renum = new LDAPSearchBuilder(context,domainDN).subTreeScope().returns("cn").search(query.toString(), sids.toArray());
-        parseMembers(userDN, groups, renum);
-        renum.close();
-
+        else {
+            // build up the query to retrieve all the groups
+            StringBuilder query = new StringBuilder("(|");
+            List<byte[]> sids = new ArrayList<byte[]>();
+    
+            NamingEnumeration<?> tokenGroups = tga.getAll();
+            while (tokenGroups.hasMore()) {
+                byte[] gsid = (byte[])tokenGroups.next();
+                query.append("(objectSid={"+sids.size()+"})");
```

Comment:
yes - but this was just an indent change - I have not changed the rest of the code so that `git blame -w` works as expected.